### PR TITLE
Rewrite the demo in TS

### PR DESCRIPTION
**Pull request by omgovich**
In order to keep the entire repo's codebase consistent.

### CODE REVIEW

File: demo/src/hooks/useFaviconColor.ts [hunk 1]
```diff
@@ -75,13 +82,17 @@ const useFaviconColor = (color) => {
...
   };
 
-  useThrottledEffect(
-    () => {
-      if (shouldReplace()) replaceFavicon(color);
-    },
-    500,
-    [color]
-  );
+  // modified from `use-throttled-effect` since the lib doesn't support TS
```
Comment (RyanChristian4427) [inline demo/src/hooks/useFaviconColor.ts:85]: If you mean by this the library does not have typings, there are easy ways around this. Don't necessarily have to write up your own solution.

One way would be to use a declaration file.

`demo/src/declaration.d.ts`
```
declare module "use-throttled-effect";
```

That'll silence TS. Obviously won't have any typings, but you can then do what you want with it. Everything should be `any`. Alternatively you could write typings for this lib here, but that's likely overkill.

Comment (omgovich) [reply demo/src/hooks/useFaviconColor.ts:85]: Yeah, I know that I could define a module, but it seems like an extra work for a supersmall thing like this hook. It's seems easier to me to just copy and and add typings.

Comment (RyanChristian4427) [reply demo/src/hooks/useFaviconColor.ts:85]: Fair enough, just wanted to make sure you knew that they're was an easier solution (albeit with drawbacks)

**Comment by omgovich** [comment]
I will close it for now since it will have too many conflicts with v3
I will come back to this project when we release a stable v3.

**Pull request closed without merging**